server/tests/replicated_loglet.rs

```diff
@@ -175,4 +182,168 @@ mod tests {
         )
         .await
     }
+
+    #[test(tokio::test)]
+    async fn bifrost_append_and_seal_concurrent() -> googletest::Result<()> {
+        const TEST_DURATION: Duration = Duration::from_secs(60);
+        const CONCURRENT_APPENDERS: usize = 20;
+
+        run_in_test_env(
+            Configuration::default(),
+            GenerationalNodeId::new(5, 1), // local sequencer
+            ReplicationProperty::new(NonZeroU8::new(2).unwrap()),
+            3,
+            |test_env| async move {
+                let log_id = LogId::new(0);
+
+                let tc = task_center();
+                let metadata = metadata();
+
+
+                let mut appenders: JoinSet<googletest::Result<_>> = JoinSet::new();
+                let cancel_appenders = CancellationToken::new();
+
+                for appender_id in 0..CONCURRENT_APPENDERS {
+                    appenders.spawn({
+                        let bifrost = test_env.bifrost.clone();
+                        let cancel_appenders = cancel_appenders.clone();
+                        let tc = tc.clone();
+                        async move {
+                            tc.run_in_scope("append", None, async move {
+                                let mut i = 1;
+                                let mut committed = Vec::new();
+                                while !cancel_appenders.is_cancelled() {
+                                    let offset = bifrost
+                                        .append(
+                                            log_id,
+                                            format!("appender-{}-record{}", appender_id, i),
+                                        )
+                                        .await?;
+                                    i += 1;
+                                    committed.push(offset);
+                                }
+                                Ok(committed)
+                            })
+                            .await
+                        }
+                    });
+                }
+
+                let mut sealer_handle: JoinHandle<googletest::Result<()>> = tokio::task::spawn({
+                    let tc = tc.clone();
+                    let (bifrost, metadata_writer, metadata_store_client) = (
+                        test_env.bifrost.clone(),
+                        test_env.metadata_writer.clone(),
+                        test_env.metadata_store_client.clone()
+                    );
+
+                    async move {
+                        let cancellation_token = cancellation_token();
+
+                        let mut chain = metadata.updateable_logs_metadata().map(|logs| logs.chain(&log_id).expect("a chain to exist"));
```

Comment:
metadata.logs_ref()).map... is probably what you actually need here. This gets you a wrapped reference of the last logs metadata.

Reply:
`metadata.logs_ref()).map` isnt a function - isnt logs_ref pinned, ie its a frozen arc from a point of time?

Reply:
it returns a reference of the last known value, and that's what you actually need here.

Reply:
hmm, but this is used in a loop, and i need the loglet id in params to change with each seal

Reply:
You are right, let's keep is as updateable in this case.